src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -717,37 +721,111 @@ class FaceBasedAssemblyKernel : public FaceBasedAssemblyKernelBase
           localIndex const esr_up = sesri[k_up];
           localIndex const ei_up  = sei[k_up];
 
-          real64 const mobility = m_phaseMob[er_up][esr_up][ei_up][ip];
-
-          // skip the phase flux if phase not present or immobile upstream
-          if( LvArray::math::abs( mobility ) < 1e-20 ) // TODO better constant
+          // C1PPU
```

Comment:
> This could be refactored in a new `FaceBasedAssemblyKernel::computeC1Flux()` that would be dispatched in the `FaceBasedAssemblyKernel::launch()` implementation based on the value of `m_useC1PPU`

If I do that, how to avoid significant code duplication between PPU and C1-PPU?

Reply:
As you will be merging first, why not create the Utilities file like here and factored out PPU like here 
https://github.com/GEOS-DEV/GEOS/blob/a50062d1af59c9e96437d62c7bb9b70ba54d2ee6/src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVMUpwindUtilities.hpp#L50-L248

Reply:
will computePPUPhaseFlux and new computeC1PPUPhaseFlux will be mostly the same?

Reply:
Actually as you don't need the all flux but just upwind direction ``k_up`` you might be able to even refactor a smaller subset and call it in ``computeFlux()`` and ``computeFluxC1PPU()``